src/main/java/network/brightspots/rcv/CastVoteRecord.java

```diff
@@ -128,6 +128,12 @@ boolean doesUseLastAllowedRanking() {
     return usesLastAllowedRanking;
   }
 
+  // This represents the canonical ID used for audit logs and RCTab CVR
+  String getPrimaryId() {
+    return !isNullOrBlank(computedId) ? computedId : suppliedId;
+  }
```

Comment:
This is a refactoring of the `computedId` and `suppliedId` logic used below in the `logStringBuilder` (which should print a unique CVR Id to the audit log.